service/frontend/validators.go

```diff
@@ -39,3 +41,47 @@ func validateStringField(fieldName string, value string, maxLen int, required bo
 	}
 	return nil
 }
+
+func validateFairnessWeightUpdate(
+	set map[string]float32,
+	unset []string,
+	maxConfigLimit int,
+) error {
+	total := len(set) + len(unset)
+	if total > maxConfigLimit {
+		return serviceerror.NewInvalidArgumentf(
+			"too many fairness weight overrides in request: got %d, maximum %d",
+			total, maxConfigLimit,
+		)
+	}
+
+	for k, w := range set {
+		if strings.TrimSpace(k) == "" {
```

Comment:
Why TrimSpace? The actual code that uses them doesn't do that, it just uses the string as-is. This should do the same.